### PR TITLE
fix CMakeToolchain warnings and werror

### LINKED ISSUE #10288: [bug] Generated CMake files cause errors with `-Werror=dev --warn-uninitialized`
Conan generated CMake files cause errors when the project is configured with `-Werror=dev --warn-uninitialized`.

### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10
  * Compiler+version: VS2019
  * Conan version: 1.44.0
  * Python version: 3.10.0

### Steps to reproduce (Include if Applicable)

`find_package()` a dependency after installing it with `CMakeToolchain` and `cmake_find_package_multi` generators.

### Logs (Executed commands with output) (Include/Attach if Applicable)

Example log attached for when using the toolchain to `find_package(freetype)`. It's just a lot of uninitialized variable use.

[log.log](https://github.com/conan-io/conan/files/7833584/log.log)

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Fix: Fix the ``CMakeToolchain`` generated code, so it doesnt fail for ``-Werror --warn-unitilized``. 
Docs: Omit

Close https://github.com/conan-io/conan/issues/10288

### CODE REVIEW

**Comment by memsharded** [comment]
After user @friendlyanon feedback, I have added checks for CMakeDeps too.

But now I find errors with ``dep_COMPONENT_NAMES`` and ``dep_FIND_DEPENDENCY_NAMES``, which requires some extra investigation, because I don't get it yet why is it complaining. 

Update: managed to solve the issue. The thing is that ``list(APPEND)`` will not initialize the variable if not appending something, but it will create and initialize de variable if previously uninitialized.

**Review by czoido — APPROVED** [review]

File: conans/test/functional/toolchains/cmake/test_cmake_toolchain.py [hunk 1]
```diff
@@ -177,3 +177,37 @@ def build(self):
...
+        class Conan(ConanFile):
+            settings = "os", "compiler", "arch", "build_type"
+            generators = "CMakeToolchain", "CMakeDeps"
+            requires = "dep/0.1"
+        """)
+    consumer = textwrap.dedent("""
+       set(CMAKE_CXX_COMPILER_WORKS 1)
+       set(CMAKE_CXX_ABI_COMPILED 1)
+       project(MyHello CXX)
+       cmake_minimum_required(VERSION 3.15)
```
Comment (friendlyanon) [inline conans/test/functional/toolchains/cmake/test_cmake_toolchain.py:200]: The order of command calls isn't right here. `cmake_minimum_required` also sets up policies according to the provided version number. The order here results in running the toolchain with no policies. The toolchain is run via `project` while enabling a language for the first time. This should be:

```cmake
cmake_minimum_required(VERSION 3.15)
set(CMAKE_CXX_COMPILER_WORKS 1)
set(CMAKE_CXX_ABI_COMPILED 1)
project(MyHello CXX)
```

Comment (memsharded) [reply conans/test/functional/toolchains/cmake/test_cmake_toolchain.py:200]: Good to know, but not critical for this PR (it has already been merged), right? The test failed without the fixes, then passed with them. It it jsut an optimization to try to save some seconds of CI time.

Comment (friendlyanon) [reply conans/test/functional/toolchains/cmake/test_cmake_toolchain.py:200]: It has been merged, yes. Just letting you know that `cmake_minimum_required` should really be the very first command call in an entrypoint.

**Pull request merged**